test/tests/performance/pbs_jobperf.py

```diff
@@ -0,0 +1,642 @@
+                      'No_of_tries': 1,
+                      'No_of_iterations': 10,
+                      'No_of_users': 10,
+                      'svr_log_level': 511,
+                      'qsub_exec': '-- /bin/true',
+                      'qsub_exec_arg': None,
+                      'No_of_moms': 21,
+                      'No_of_ncpus_per_node': 48
+        """
+        testconfig = {'No_of_jobs_per_user': 100,
+                      'No_of_tries': 1,
+                      'No_of_iterations': 10,
+                      'No_of_users': 10,
+                      'svr_log_level': 511,
+                      'qsub_exec': '-- /bin/true',
+                      'qsub_exec_arg': None,
+                      'No_of_moms': 21,
+                      'No_of_ncpus_per_node': 48}
+        config = self.set_test_config(testconfig)
+
+        avg_sub_time = []
+        avg_run_rate = []
+        avg_throughput = []
+        sub_time = []
+        run_rate = []
+        throughput = []
+        j = 0
+        nodes = self.server.status(NODE, 'state=free')
+        if len(nodes) < config['No_of_moms']:
+            a = {'resources_available.ncpus': config['No_of_ncpus_per_node']}
+            self.server.create_moms('mom', a, config['No_of_moms'], self.mom)
+
+        while j < config['No_of_tries']:
+            i = 0
+            log_start = time.time()
+            users = ['pbsuser1', 'pbsuser2', 'pbsuser3', 'pbsuser4',
+                     'pbsuser5', 'pbsuser6', 'pbsuser7', 'pbstest',
+                     'pbsroot', 'pbsadmin']
+            a = {'log_events': config['svr_log_level']}
+            self.server.manager(MGR_CMD_SET, SERVER, a, expect=True)
+            while i < config['No_of_iterations']:
+                os.chdir('/tmp')
+                thrds = []
+                start = time.time()
+                for u in range(0, config['No_of_users']):
+                    t = multiprocessing.Process(target=self.submit_jobs, args=(
+                        users[u], config['No_of_jobs_per_user'],
+                        config['qsub_exec'],
+                        config['qsub_exec_arg']))
+                    t.start()
+                    thrds.append(t)
+                for t in thrds:
+                    t.join()
+                i = i + 1
+                stop = time.time()
+                res = stop - start
+                resps = (config['No_of_jobs_per_user']
+                         * config['No_of_users']) / res
+                sub_time.append(resps)
+            j += 1
+
+            sclg = PBSLogAnalyzer()
+            md = sclg.analyze_scheduler_log(
+                filename=self.scheduler.logfile, start=int(log_start))
+            jr = md['summary']['num_jobs_run']
+            total_jobs = config['No_of_jobs_per_user']
+            total_jobs *= config['No_of_users']
+            total_jobs *= config['No_of_iterations']
+            total_jobs *= config['No_of_tries']
+            while jr != total_jobs:
```

Comment:
wont job_state=F only work when history is enabled ?

Reply:
@visheshh  I don't see a reason why you can not have history enabled. You don't have to run analyzer in a loop if you do that.
Else you could also check for 0 queued/running jobs in the system if you want.

Reply:
@arungrover I made the changes . But when it runs qstat and after every qstat it turns scheduling=t and kicks a cycle does it not affect throughput of jobs?

Reply:
Since most of the jobs are /bin/true, chances are scheduler would anyways be in cycle when qstat sets scheduling. But in your case, you have a use case where qstat shouldn't trigger a cycle. So to address this I would change framework and add an option to status call so that it does not trigger any cycles, the option by default will be true so that no other test is affected and your test can pass it as False.
What do you think?

Reply:
Sounds good , Thanks for the suggestion .